Source/WebCore/page/Page.cpp

```diff
@@ -3794,6 +3815,33 @@ void Page::setFullscreenControlsHidden(bool hidden)
 #endif
 }
 
+Document* Page::outermostFullscreenDocument() const
+{
+#if ENABLE(FULLSCREEN_API)
+    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_mainFrame.get());
```

Comment:
Does anything in the method require smart pointers?

Reply:
As a `const` method, it should _in theory_ not mutate anything in such a way that `m_mainFrame` changes during the runtime of the method. In practice, calling callbacks and dispatching events make that not true, but neither of those things happens here.

Reply:
I'll modify all these raw pointers to use CheckedPtr however.